File: LivrusVirtus/FoxToolKit/FXPython/CadPaises.py
from FXPy.fox import *
from Cadastros import frmCadastros
from resource import Resource
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
import Principal

class frmCadPaises(frmCadastros):

  def __init__(self, clienteMDI, iconeMDI, menuMDI):  
    frmCadastros.__init__(self, clienteMDI, Resource.STR_CADPAIS, 
      iconeMDI, menuMDI, TEXTFIELD_NORMAL, True)

    self.Rotinas = RotinasGlobais.Rotinas
    self.ConsSQL = ConsultasSQL.ConsSQL

    FXMAPFUNC(self, SEL_COMMAND, frmCadPaises.ID_FRMCADASTROS_BTNNOVO, 
      frmCadPaises.onBtnNovo)
    FXMAPFUNC(self, SEL_COMMAND, frmCadPaises.ID_FRMCADASTROS_BTNPESQUISAR, 
      frmCadPaises.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadPaises.ID_FRMCADASTROS_BTNANTERIOR, 
      frmCadPaises.onBtnAnterior)
    FXMAPFUNC(self, SEL_COMMAND, frmCadPaises.ID_FRMCADASTROS_BTNPROXIMO, 
      frmCadPaises.onBtnProximo)
    FXMAPFUNC(self, SEL_COMMAND, frmCadPaises.ID_FRMCADASTROS_BTNSALVAR, 
      frmCadPaises.onBtnSalvar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadPaises.ID_FRMCADASTROS_BTNEXCLUIR, 
      frmCadPaises.onBtnExcluir)
    FXMAPFUNC(self, SEL_COMMAND, frmCadPaises.ID_FRMCADASTROS_BTNLIMPAR, 
      frmCadPaises.onBtnLimpar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadPaises.ID_FRMCADASTROS_BTNFECHAR, 
      frmCadPaises.onBtnFechar)
    FXMAPFUNC(self, SEL_CLOSE, 0, frmCadPaises.onClose)

    self.labDescricao.setText("Pa\xeds: ")

  def ValidaDados(self, bTodosDados):
    if self.Rotinas.ValidaCampo(self.edtCodigo.getText(), Resource.STR_CODNINF):
   	  return False
    if bTodosDados:
  	  if self.Rotinas.ValidaCampo(self.edtDescricao.getText(), Resource.STR_PAISNINF):
    		return False        	
    return True
  
  def onBtnNovo(self, sender, sel, ptr):
    self.btnNovoClick()
    return True  
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(
      self.ConsSQL.Pais('S', self.edtCodigo.getText(),""), Resource.STR_PAISNENC)
    return True  
  
  def onBtnAnterior(self, sender, sel, ptr):
    self.registroAnterior()	
    return True  
  
  def onBtnProximo(self, sender, sel, ptr):
    self.registroProximo()
    return True  
  
  def onBtnSalvar(self, sender, sel, ptr):
    self.SalvarDados(self.ValidaDados(True),
      self.ConsSQL.Pais('S', self.edtCodigo.getText(),""),
      self.ConsSQL.Pais('U', self.edtCodigo.getText(), 
        self.edtDescricao.getText()),
      self.ConsSQL.Pais('I', self.edtCodigo.getText(), 
        self.edtDescricao.getText()), 
      True)
    return True  
  
  def onBtnExcluir(self, sender, sel, ptr):
    self.ExcluirDados(self.ValidaDados(False),
      self.ConsSQL.Pais('D', self.edtCodigo.getText(),""))
    return True  
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.btnLimparClick()
    return True  

  def onBtnFechar(self, sender, sel, ptr):
    self.onClose(sender, sel, ptr)
    return True
  
  def onClose(self, sender, sel, ptr):
    Principal.fPrincipal.habilitaMenu("CadPaises")
    self.destroy()
    return True

File: LivrusVirtus/FoxToolKit/FXRuby/CadPaises.rb
require 'fox'
require 'fox/responder'
require 'Cadastros'
require 'Principal'
require 'RotinasGlobais'
require 'ConsultasSQL'
include Fox
include Responder

class FRMCadPaises < FRMCadastros
  
  def initialize(clienteMDI, iconeMDI, menuMDI)  
    super(clienteMDI, Resource::STR_CADPAIS, iconeMDI, menuMDI, 
      TEXTFIELD_NORMAL, true)

    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNNOVO, :onBtnNovo)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNPESQUISAR, :onBtnPesquisar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNANTERIOR, :onBtnAnterior)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNPROXIMO, :onBtnProximo)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNSALVAR, :onBtnSalvar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNEXCLUIR, :onBtnExcluir)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNLIMPAR, :onBtnLimpar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNFECHAR, :onBtnFechar)
    FXMAPFUNC(SEL_CLOSE, 0, :onClose)

    @edtCodigo.setNumColumns(3)
    @labDescricao.setText("País: ")
  end

  def ValidaDados(bTodosDados)
    if $Rotinas.ValidaCampo(@edtCodigo.getText(), Resource::STR_CODNINF)
    	  return false
    end
  	if bTodosDados
  	  if $Rotinas.ValidaCampo(@edtDescricao.getText(), Resource::STR_PAISNINF)
    		return false
      end
  	end
  	return true
  end 
  
  def onBtnNovo(sender, sel, ptr)
    btnNovoClick()
    return true
  end
  
  def onBtnPesquisar(sender, sel, ptr)
    PesquisarDados(
      $ConsSQL.Pais('S', @edtCodigo.getText(),""), Resource::STR_PAISNENC)
    return true
  end
  
  def onBtnAnterior(sender, sel, ptr)
    registroAnterior()	
    return true
  end
  
  def onBtnProximo(sender, sel, ptr)
    registroProximo()
    return true
  end
  
  def onBtnSalvar(sender, sel, ptr)
    SalvarDados(ValidaDados(true),
      $ConsSQL.Pais('S', @edtCodigo.getText(),""),
      $ConsSQL.Pais('U', @edtCodigo.getText(), @edtDescricao.getText()),
      $ConsSQL.Pais('I', @edtCodigo.getText(), @edtDescricao.getText()), 
      true)
    return true
  end
  
  def onBtnExcluir(sender, sel, ptr)
    ExcluirDados(ValidaDados(false),
      $ConsSQL.Pais('D', @edtCodigo.getText(),""))
    return true
  end
  
  def onBtnLimpar(sender, sel, ptr)
    btnLimparClick()
    return true
  end

  def onBtnFechar(sender, sel, ptr)
    onClose(sender, sel, ptr)
    return true
  end

  def onClose(sender, sel, ptr)
    $frmPrincipal.habilitaMenu("CadPaises")
    destroy()
    return true
  end 
 
end
